File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    ans = []
    dqp = deque(progresses)
    dqs = deque(speeds)
    while len(dqp)>0:
        cnt=0
        while len(dqp)>0 and dqp[0]>=100:
            dqp.popleft()
            dqs.popleft()
            cnt+=1
        if cnt>0:
            ans.append(cnt)
        for i in range(len(dqp)):
            dqp[i] += dqs[i]
    return ans

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let ans = []
    while(progresses.length>0){
        for(let i=0;i<progresses.length;i++){
            progresses[i]+=speeds[i]
        }
        let cnt = 0
        while(progresses[0]>=100){
            progresses.shift()
            speeds.shift()
            cnt++
        }
        if(cnt>0) ans.push(cnt)
    }
    return ans
}